### ISSUE TITLE
[clippy::missing-panics-doc] `todo!()` triggers lint

### ISSUE THREAD

**Issue by Starwort** [rust-lang/rust-clippy#10966]
### Summary

When linting for `clippy::missing-panics-doc` (e.g. under pedantic), the lint is triggered for `todo!()`s in the code. I believe this is unintended as `todo!()` is a placeholder for unfinished code; the code replacing `todo!()` is likely to not panic, and adding a 'Panics' section likely to be removed once the function is necessary seems unnecessary.

### Lint Name

clippy::missing-panics-doc

### Reproducer

I tried this code:

```rust
/// This is my awesome API function, that does X, Y, and Z
pub fn not_finished() {
    // some logic
    todo!()
}
```

I saw this happen:

A lint warning was issued for the function (`clippy::missing-panics-doc`) due to the `todo!()`

I expected to see this happen:

No lint warning as the code is incomplete, and does not contain any other source of a panic

### Version

```text
rustc 1.72.0-nightly (498553fc0 2023-05-29)
binary: rustc
commit-hash: 498553fc04f6a3fdc53412320f4e913bc53bc267
commit-date: 2023-05-29
host: x86_64-unknown-linux-gnu
release: 1.72.0-nightly
LLVM version: 16.0.4
```

**Comment by dswij** [comment]
This lint intentionally triggers for `todo!()` ([see](https://github.com/rust-lang/rust-clippy/blame/master/tests/ui/missing_panics_doc.rs#L17)).

And `todo!()` does panic when called. Although, from a user perspective, it's going to be a tad annoying. a WIP might have a lot of todo and Clippy will complains a lot on the way. We have [another lint](https://rust-lang.github.io/rust-clippy/master/index.html#/todo) for `todo!()` anyways, so I guess excluding `todo!()` makes sense.

**Comment by xFrednet** [comment]
The lint triggering on `todo!` macros has been annoying me for some time. I'm in favor of excluding it, or at least adding a config value to disable it. :)
